cirq/neutral_atoms/neutral_atom_devices_test.py

```diff
@@ -0,0 +1,310 @@
+        d.validate_gate(cirq.CNotPowGate(exponent=0.5))
+    assert 'controlled gates must have integer exponents' == str(bad_exp.value)
+    with pytest.raises(ValueError) as bad_gate:
+        d.validate_gate(cirq.SingleQubitGate())
+    assert 'Unsupported gate' in str(bad_gate.value)
+
+
+def test_validate_operation_errors():
+    d = square_device(3, 3)
+
+    class bad_op(cirq.Operation):
+
+        def bad_op(self):
+            pass
+
+        def qubits(self):
+            pass
+
+        def with_qubits(self, new_qubits):
+            pass
+
+    with pytest.raises(ValueError) as bad_operation:
+        d.validate_operation(bad_op())
+    assert "Unsupported operation" in str(bad_operation.value)
+    not_on_device_op = cirq.ParallelGateOperation(cirq.X,
+                                                  [cirq.GridQubit(row, col)
+                                                   for col in range(4)
+                                                   for row in range(4)])
+    with pytest.raises(ValueError) as bad_qubits:
+        d.validate_operation(not_on_device_op)
+    assert "Qubit not on device" in str(bad_qubits.value)
+    with pytest.raises(ValueError) as control_limit:
+        d.validate_operation(cirq.CCX.on(*d.qubit_list()[0:3]))
+    assert "Too many qubits acted on in parallel by" in str(control_limit.value)
+    with pytest.raises(ValueError) as too_far:
+        d.validate_operation(cirq.CZ.on(cirq.GridQubit(0, 0),
+                                        cirq.GridQubit(2, 2)))
+    assert "are too far away" in str(too_far.value)
+    with pytest.raises(ValueError) as too_many_z:
+        d.validate_operation(cirq.ParallelGateOperation(cirq.Z, d.qubits))
+    assert "Too many Z gates in parallel" == str(too_many_z.value)
+    with pytest.raises(ValueError) as bad_xy_num:
+        d.validate_operation(cirq.ParallelGateOperation(cirq.X,
+                                                        d.qubit_list()[1:]))
+    assert "Bad number of XY gates in parallel" == str(bad_xy_num.value)
+
+
+def test_validate_moment_errors():
+    d = square_device(3, 3)
+    q00 = cirq.GridQubit(0, 0)
+    q01 = cirq.GridQubit(0, 1)
+    q10 = cirq.GridQubit(1, 0)
+    q11 = cirq.GridQubit(1, 1)
+    q12 = cirq.GridQubit(1, 2)
+    q02 = cirq.GridQubit(0, 2)
+    q04 = cirq.GridQubit(0, 4)
+    q03 = cirq.GridQubit(0, 3)
+    q20 = cirq.GridQubit(2, 0)
+    q21 = cirq.GridQubit(2, 1)
+
+    m = cirq.Moment([cirq.Z.on(q00), (cirq.Z**2).on(q01)])
+    with pytest.raises(ValueError) as non_id_z:
+        d.validate_moment(m)
+    assert "Non-identical simultaneous " in str(non_id_z.value)
+    m = cirq.Moment([cirq.X.on(q00), cirq.Y.on(q01)])
+    with pytest.raises(ValueError) as non_id_xy:
+        d.validate_moment(m)
+    assert "Non-identical simultaneous " in str(non_id_xy.value)
+    m = cirq.Moment([cirq.CNOT.on(q00, q01), cirq.CZ.on(q12, q02)])
+    with pytest.raises(ValueError) as non_id_c:
```

Comment:
Use `pytest.raises(ValueError, match="Non-identical simultaneous")`, instead of getting the exception and asserting on its str. Repeat for all other checks of this type.

Reply:
I now use match on all checks to verify whether the error that was thrown was indeed the one I expected to be thrown.